partner_manual_rank/models/res_partner.py

```diff
@@ -25,12 +25,14 @@ class ResPartner(models.Model):
     @api.depends("customer_rank")
     def _compute_is_customer(self):
         for partner in self:
-            partner.is_customer = bool(partner.customer_rank)
+            if not partner.is_customer:
+                partner.is_customer = bool(partner.customer_rank)
 
     @api.depends("supplier_rank")
     def _compute_is_supplier(self):
         for partner in self:
-            partner.is_supplier = bool(partner.supplier_rank)
+            if not partner.is_supplier:
```

Comment:
What if you are unchecking the is_supplier/is_customer boolean? does this work?

Reply:
Yes, because when you change it manually the compute method is not called (this field have the readonly flag as False)